app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileEdit.kt

```diff
@@ -1,215 +1,229 @@
+                        Modifier.fillMaxWidth().height(descriptionHeight).testTag("description"))
+                Spacer(modifier = Modifier.height(padding))
+
+                TagSelector(
+                    selectedTag =
+                        selectedTags.map { tag -> Pair(tag, tagsViewModel.tagToColor(tag)) },
+                    outputTag =
+                        tagsSuggestions.value.map { tag ->
+                          Pair(tag, tagsViewModel.tagToColor(tag))
+                        },
+                    stringQuery = stringQuery.value,
+                    expanded = expanded,
+                    onTagClick = { tag -> tagsViewModel.removeFilter(tag) },
+                    onStringChanged = { tagsViewModel.setQuery(it, selectedTags) },
+                    textColor = MaterialTheme.colorScheme.onSurface,
+                    onDropDownItemClicked = { tag -> tagsViewModel.addFilter(tag) },
+                    height = screenHeight,
+                    width = screenWidth,
+                    textSize = (screenHeight.value * 0.03).sp,
+                )
+
+                Spacer(modifier = Modifier.height(padding))
+
+                // Modal for unsaved changes on leave page with back button
+                if (showDialogBack) {
+                  AlertDialog(
+                      onDismissRequest = { showDialogBack = false },
+                      title = { Text("You are about to leave this page") },
+                      text = { Text("Your changes will not be saved.") },
+                      confirmButton = {
+                        TextButton(
+                            onClick = {
+                              showDialogBack = false
+                              tagsViewModel.leaveUI()
+                              navigationActions.goBack()
+                            }) {
+                              Text("Discard changes")
+                            }
+                      },
+                      dismissButton = {
+                        TextButton(onClick = { showDialogBack = false }) { Text("Cancel") }
+                      },
+                      modifier = Modifier.testTag("alertDialogBack"))
+                } else if (showDialogConfirm) {
+                  AlertDialog(
+                      onDismissRequest = { showDialogConfirm = false },
+                      title = { Text("You are about to leave this page") },
+                      text = { Text("Are you sure you want to save your changes?") },
+                      confirmButton = {
+                        TextButton(
+                            onClick = {
+                              showDialogConfirm = false
+                              updateProfile()
+                              tagsViewModel.leaveUI()
+                              navigationActions.goBack()
+                            }) {
+                              Text("Save")
+                            }
+                      },
```

Comment:
Design: Even if having a dialog to confirm data update, it doesn't feel right when using the app. I think adding a prompt to confirm the changes adds too much friction to the edit process and is not required, as a user can always re-edit it's profile if something is missing or wrong.